drivers/modem/quectel-bg9x.c

```diff
@@ -1029,8 +1029,29 @@ static int modem_setup(void)
 			     NULL, 0U, "AT+QIACT=1", &mdata.sem_response,
 			     MDM_CMD_TIMEOUT);
 
-	/* Retry or Possibly Exit. */
-	if (ret < 0 && init_retry_count++ < MDM_INIT_RETRY_COUNT) {
+	/* If there is trouble opening the PDP context, we try to deactivate/reactive it 3 times. */
+	while (ret == -EIO && pdp_act_retry_count < MDM_PDP_ACT_RETRY_COUNT) {
+
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, "AT+QIDEACT=1", &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+		/* If there's an error for the AT+QIDEACT, restart the module. */
+		if (ret < 0) {
+			goto restart;
+		}
+
+		ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, "AT+QIACT=1", &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+		pdp_act_retry_count++;
+	}
```

Comment:
`modem_setup()` is already too big. Please create a helper function like `bg9x_pdp_context_open()` or something and put the logic related to open/close retries there. This by the way will make `pdp_act_retry_count` also count from 0. This is not true in current implementation, because above `goto restart` doesn't do that (and I think it should).

Reply:
It it is rather "activate/deactivate context" than "open/close context". Let's use one or the other. I think "activate/deactivate" is better and fits AT+QIACT=1 and AT+QIDEACT=1.

Reply:
Fixed.